### ISSUE TITLE
Migrating to 5.x get ClassCastException on StringTerms

### ISSUE THREAD

**Issue by 123avi** [sksamuel/elastic4s#826]
I ma trying do this aggregation (worked on 2.x) :
```
def getValuesByField(tag: String){
      client execute {
search(indexName / indexType) query ("_field_names", tag) aggregations (termsAggregation( "agg") field s"$tag.fields")
}.map { r =>
      r.aggregations.stringTermsResult("agg").getBuckets.toList.map(_.getKey.toString)
}
```
but I am getting this crash

> java.lang.ClassCastException: org.elasticsearch.search.aggregations.bucket.terms.UnmappedTerms cannot be cast to org.elasticsearch.search.aggregations.bucket.terms.StringTerms
> 	at com.sksamuel.elastic4s.searches.aggs.RichAggregations.stringTermsResult(RichAggregations.scala:33)

All mapped values are strings

**Comment by sksamuel** [comment]
Which version of 5.2 are you on ?

**Comment by 123avi** [comment]
5.2.11

**Comment by sksamuel** [comment]
This is fixed in the 6.0.0 http client release.
